.github/workflows/ion-java-performance-regression-detector.yml

```diff
@@ -21,31 +21,50 @@ jobs:
+      - name: Checkout the ion-java from the existing commit.
         uses: actions/checkout@v2
         with:
           fetch-depth: 2
           submodules: recursive
-      - run: git checkout HEAD^
+          run: git checkout HEAD^
 
-      - name: Build ion-java from the current commit
+      - name: Build ion-java based on the existing commit
         run: mvn clean install
 
-      - name: Build ion-java-benchmark-cli based on the current ion-java
+      - name: Checkout ion-java-benchmark-cli
         uses: actions/checkout@v2
         with:
-          repository: amzn/ion-java-benchmark-cli
-          ref: master
-      - run: mvn clean install
+          repository: linlin-s/ion-java-benchmark-cli
+          ref: final
+
+      - name: Build ion-java-benchmark-cli based on the current ion-java
+        run: mvn clean install
 
       - name: Check the version of ion-java
         run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar --version
 
       - name: Generate test Ion Data
-        run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar generate -S 500 -T string -f ion_text testWorkflow.ion
+        run: |
+          mkdir -p testData
+          java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar  generate -S 5000 --input-ion-schema ./tst/com/amazon/ion/benchmark/testStruct.isl /home/runner/work/ion-java/ion-java/testData/testStruct.10n
+          java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar  generate -S 5000 --input-ion-schema ./tst/com/amazon/ion/benchmark/testList.isl /home/runner/work/ion-java/ion-java/testData/testList.10n
+          java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar  generate -S 5000 --input-ion-schema ./tst/com/amazon/ion/benchmark/testNestedStruct.isl /home/runner/work/ion-java/ion-java/testData/testNestedStruct.10n
+      - name: Upload test Ion Data to artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: test Ion Data
+          path: testData
 
-      - name: Test read preformance of the ion-java from the current commit
-        run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar read testWorkflow.ion -o readPerformanceCurrent.ion -r ion
+      - name: Benchmark ion-java from the existing commit
+        run: |
+          mkdir -p benchmarkResults
+          java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar organize --test-ion-data /home/runner/work/ion-java/ion-java/testData /home/runner/work/ion-java/ion-java/benchmarkResults
+
+      - name: Upload benchmark results to artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: Benchmark result
+          path: benchmarkResults
 
       - name: Clean maven dependency repository
         run : rm -r /home/runner/.m2
```

Comment:
> This comment is about [line72](https://github.com/linlin-s/ion-java/blob/f45d46f2c3a3e518131e16b4f704364d67b1982c/.github/workflows/ion-java-performance-regression-detector.yml#L72),
> 
> We only cleaned up the maven dependencies above but not ion-java right? So it might not necessary to checkout the ion-java again. Do you think we could just `git reset` to the commit we want and then `mvn clean install` instead?

Yes, that's a good idea, changes will be updated in the following commit.